src/MOI/MOI_wrapper.jl

```diff
@@ -2801,21 +2818,34 @@ function MOI.get(model::Optimizer, attr::MOI.DualStatus)
     end
 end
 
+_update_cache(::Optimizer, data::Vector{Float64}, ::Any) = data
+
+function _update_cache(model::Optimizer, ::Nothing, f_p::F) where {F}
+    n = length(model.variable_info)
+    x = zeros(n)
+    ret = f_p(model.env, model.lp, x, 0, n-1)
+    _check_ret(model, ret)
+    return x
+end
+
+_get_cached_solution(model::Optimizer, data::Vector{Float64}, x) = data[column(model, x)]
+
 function MOI.get(
     model::Optimizer,
     attr::MOI.VariablePrimal,
-    x::MOI.VariableIndex,
+    x::Union{MOI.VariableIndex,Vector{MOI.VariableIndex}},
 )
     _throw_if_optimize_in_progress(model, attr)
     MOI.check_result_index_bounds(model, attr)
-    col = Cint(column(model, x) - 1)
     if model.has_primal_certificate
-        return model.certificate[col+1]
+        return model.certificate[column(model, x)]
     end
-    x = Ref{Cdouble}()
-    ret = CPXgetx(model.env, model.lp, x, col, col)
-    _check_ret(model, ret)
-    return x[]
+    model.variable_primal = _update_cache(
+        model,
+        model.variable_primal,
+        CPXgetx,
+        )
```

Comment:
Did you run JuliaFormatter locally? Not really important, but it helps maintain the consistency of the code.

Reply:
@Gbrunieau you can see the formatting failures here: https://github.com/jump-dev/CPLEX.jl/runs/4169362806?check_suite_focus=true

Reply:
thanks I didn't know about JuliaFormatter, I will use it from now on.